media/server/gstplayer/source/GstGenericPlayer.cpp

```diff
@@ -498,6 +498,62 @@ GstElement *GstGenericPlayer::getDecoder(const MediaSourceType &mediaSourceType)
     return nullptr;
 }
 
+GstElement *GstGenericPlayer::getFilter(const MediaSourceType &mediaSourceType)
```

Comment:
This functions searches for parser, not filter from gstreamer's classification perspective.
What if you would have multiple PARSER | MEDIA_SOURCE elements in the pipeline? It would find only the first element that matches the type. Maybe it would be better to find element that matches the given type AND has to searches property? OR function, which returns all elements of given type in the pipeline.

Reply:
there is no such thing like TYPE_FILTER in gstreamer. Filters have Codec/Parser type.

Reply:
function name changed to getParser